### DISCUSSION TITLE
Creating new device service

### DISCUSSION THREAD

**Discussion by JOGIMANISHA0706** [edgexfoundry/community#149] [Q&A]
How to create a new device service for watch dog timer in C language ? please help in doing this.

**Comment by FelixTing** [comment]
Hi @JOGIMANISHA0706, to write a device service using C SDK, you can refer to the C SDK Getting Started Guide: https://docs.edgexfoundry.org/3.1/getting-started/Ch-GettingStartedSDK-C/

**Comment by JOGIMANISHA0706** [comment]
Thank you for your email.

I've already gone through the link that you sent. But for a particular
device like Watch Dog timer I need.

Can you please check and share your knowledge with me.

**Reply by cloudxxx8** [reply]
We don't have any knowledge about the particular device.  You might contact the device vendor to understand the protocol spec and write the protocol driver logic.

**Comment by JOGIMANISHA0706** [comment]
Thank you for your response.

Can you please elaborate how to connect with device vendor !.

**Reply by cloudxxx8** [reply]
It means the people or company who sell you the devices.  They know how the protocol works.
